src/contracts/core/ReleaseManager.sol

```diff
@@ -0,0 +1,193 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity ^0.8.27;
+
+import {OwnableUpgradeable} from "./OwnableUpgradeable.sol";
+import {ReentrancyGuard} from "@openzeppelin/contracts/security/ReentrancyGuard.sol";
+import {Checkpoints} from "@openzeppelin/contracts/utils/structs/Checkpoints.sol";
+
+import {IReleaseManager} from "../interfaces/IReleaseManager.sol";
+import {IPermissionController} from "../interfaces/IPermissionController.sol";
+import {ReleaseManagerStorage} from "./ReleaseManagerStorage.sol";
+
+/**
+ * @title ReleaseManager
+ * @author Your Organization
+ * @notice Contract for managing an append-only log of OCI artifact releases for AVS deployments.
+ */
+contract ReleaseManager is ReentrancyGuard, OwnableUpgradeable, ReleaseManagerStorage {
+    /**
+     *
+     *                         MODIFIERS
+     *
+     */
+
+    modifier onlyRegistered(address avs) {
+        if (!registeredAVS[avs]) revert AVSNotRegistered();
+        _;
+    }
+
+    modifier onlyAuthorized(address avs) {
+        if (!_isAuthorized(avs, msg.sender)) revert Unauthorized();
+        _;
+    }
+
+    /**
+     *
+     *                         INITIALIZATION
+     *
+     */
+
+    function initialize(address _permissionController) public initializer {
+        __Ownable_init();
+        permissionController = IPermissionController(_permissionController);
```

Comment:
Setting of the PermissionController should happen in the constructor and it should be set as an `immutable` so that it can't be changed later.

https://solidity-by-example.org/immutable/